pymc4/distributions/continuous.py

```diff
@@ -1232,54 +1233,69 @@ def upper_limit(self):
-#         plt.ylim(0, 2.5)
-#         plt.legend(loc=1)
-#         plt.show()
-
-#     ========  ====================================================
-#     Support   :math:`x \in [0, \infty)`
-#     Mean      :math:`\beta \Gamma(1 + \frac{1}{\alpha})`
-#     Variance  :math:`\beta^2 \Gamma(1 + \frac{2}{\alpha} - \mu^2)`
-#     ========  ====================================================
-
-#     Parameters
-#     ----------
-#     concentration : float|tensor
-#         Shape parameter (concentration > 0).
-#     scale : float|tensor
-#         Scale parameter (scale > 0).
-#     """
-
-#     def __init__(self, name, concentration, scale, **kwargs):
-#         super().__init__(name, concentration=concentration, scale=scale, **kwargs)
-
-#     @staticmethod
-#     def _init_distribution(conditions):
-#         concentration, scale = conditions["concentration"], conditions["scale"]
-#         return tfd.Weibull(concentration=concentration, scale=scale)
+class Weibull(PositiveContinuousDistribution):
+    r"""Weibull random variable.
+
+    The pdf of this distribution is
+
+    .. math::
+
+       f(x \mid \alpha, \beta) =
+           \frac{\alpha x^{\alpha - 1}
+           \exp(-(\frac{x}{\beta})^{\alpha})}{\beta^\alpha}
+
+    .. plot::
+
+        import matplotlib.pyplot as plt
+        import numpy as np
+        import scipy.stats as st
+        plt.style.use('seaborn-darkgrid')
+        x = np.linspace(0, 3, 200)
+        concentrations = [.5, 1., 1.5, 5., 5.]
+        scales = [1., 1., 1., 1.,  2]
+        for concentration, scale in zip(concentrations, scales):
+            pdf = st.weibull_min.pdf(x, concentration, scale)
+            plt.plot(x, pdf, label=r'$\alpha$ = {}, $\beta$ = {}'.format(concentration, scale))
+        plt.xlabel('x', fontsize=12)
+        plt.ylabel('f(x)', fontsize=12)
+        plt.ylim(0, 2.5)
+        plt.legend(loc=1)
+        plt.show()
+
+    ========  ====================================================
+    Support   :math:`x \in [0, \infty)`
+    Mean      :math:`\beta \Gamma(1 + \frac{1}{\alpha})`
+    Variance  :math:`\beta^2 \Gamma(1 + \frac{2}{\alpha} - \mu^2)`
+    ========  ====================================================
+
+    Parameters
+    ----------
+    low : float|tensor
+        Lower limit of underlying uniform distribution (Use 0 for standard Weibull distribution). 
+    high : float|tensor
+        Upper limit of underlying uniform distribution (Use 1 for standard Weibull distribution).
+    concentration : float|tensor
+        Shape parameter (concentration > 0).
+    scale : float|tensor
+        Scale parameter (scale > 0).
+    """
+
+    def __init__(self, name, low, high, concentration, scale, **kwargs):
+        super().__init__(
+            name, low=low, high=high, concentration=concentration, scale=scale, **kwargs
+        )
+
+    @staticmethod
+    def _init_distribution(conditions):
+
+        low, high, concentration, scale = (
```

Comment:
The user must not pick either `low` or `high`. You must set them as `tf.zeros(shape)` and `tf.ones(shape)` respectively. The `shape` must be equal to the shape to which `concentration` and `scale` will broadcast to.

Reply:
I have used `tf.zeros_like()` and `tf.ones_like()` and used the shape of `scale`

Reply:
I think that this isn't enough. Image if the user speficies a scalar `scale` but a vector `concentration`. This will run into exactly the same issue you were seeing before when you passed scalar values to `low` and `high`. What you have to do, is to look at the `shape` that would come out of broadcasting `scale` and `concentration`, and use that `shape` with `tf.zeros` and `tf.ones`.

Reply:
shape = tf.broadcast_static_shape(tf.constant(concentration).shape,tf.constant(scale).shape)

low = tf.zeros(shape)
high = tf.ones(shape)

Does this look like a good workaround?

Reply:
No, don't use `tf.constant`, use `tf.convert_to_tensor` instead. Then, I'm not really sure if these will be static shapes, maybe @junpenglao could comment on this?

Reply:
There's also `tf.broadcast_dynamic_shape` https://www.tensorflow.org/api_docs/python/tf/broadcast_dynamic_shape

Reply:
I tried to dig into how tfp does its broadcasting and I found that they [defined some helper functions](https://github.com/tensorflow/probability/blob/a914215150d14f2a3a4e3e5f554435e36a847e2a/tensorflow_probability/python/internal/distribution_util.py#L971-L1052) to broadcast tensors or to simply get their ranks and shapes, preferring the static shapes over the dynamic ones. You could import and use those.